src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -2471,9 +2472,16 @@ public override BoundNode VisitUsingStatement(BoundUsingStatement node)
 
             if (_trackExceptions) NotePossibleException(node);
             VisitStatement(node.Body);
+
+            if (AsyncUsingAddsPendingBranch && node.AwaitOpt != null)
+            {
+                _pendingBranches.Add(new PendingBranch(node, this.State));
+            }
             return null;
         }
 
+        public abstract bool AsyncUsingAddsPendingBranch { get; }
```

Comment:
>AsyncUsingAddsPendingBranch [](start = 29, length = 27)

I am not sure why do we need this for using, but regular await expressions don't use similar mechanism.

Reply:
The reason is that `ControlFlowPass` doesn't dig into expressions, so it doesn't add pending branches for `await`s (implemented in base class).
I want to get the same behavior for async-using, but without cloning the implementation of `VisitUsingStatement`.

```C#
// from ControlFlowPass
        public override BoundNode Visit(BoundNode node)
        {
            // there is no need to scan the contents of an expression, as expressions
            // do not contribute to reachability analysis (except for constants, which
            // are handled by the caller).
            if (!(node is BoundExpression))
            {
                return base.Visit(node);
            }

            return null;
        }
```

```C#
// from PreciseAbstractFlowPass<LocalState> 
        public override BoundNode VisitAwaitExpression(BoundAwaitExpression node)
        {
            VisitRvalue(node.Expression);
            _pendingBranches.Add(new PendingBranch(node, this.State));
            if (_trackExceptions) NotePossibleException(node);
            return null;
        }
```